src/Controllers/GoalController.cs

```diff
@@ -0,0 +1,122 @@
+            return CreatedAtAction(nameof(GetActiveGoals), new { userId }, response);
+        }
+        catch (Exception e)
+        {
+            return HandleException(e);
+        }
+    }
+
+    [HttpPut("{userId}/{goalId}")]
+    [ProducesResponseType(typeof(ResponseGoal), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> UpdateGoal(Guid userId, Guid goalId, [FromBody] CreateGoal goal)
+    {
+        try
+        {
+            var updatedGoal = await _goalService.UpdateGoalAsync(userId, goalId, goal);
+            var response = ResponseGoal.FromDomain(updatedGoal);
+            return Ok(response);
+        }
+        catch (Exception e)
+        {
+            return HandleException(e);
+        }
+    }
+
+    [HttpGet("{userId}/active")]
+    [ProducesResponseType(typeof(IEnumerable<ResponseGoal>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetActiveGoals(Guid userId)
+    {
+        try
+        {
+            var activeGoals = await _goalService.GetActiveGoalsAsync(userId);
+            var response = activeGoals.Select(ResponseGoal.FromDomain);
+            return Ok(response);
+        }
+        catch (Exception e)
+        {
+            return HandleException(e);
+        }
+    }
+
+    [HttpDelete("{userId}/{goalId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> DeleteGoal(Guid userId, Guid goalId)
+    {
+        try
+        {
+            var result = await _goalService.DeleteGoalAsync(goalId, userId);
+            if (!result)
+            {
+                return NotFound("Goal not found or belongs to another user.");
+            }
+            return Ok("Goal deleted successfully.");
+        }
+        catch (Exception e)
+        {
+            return HandleException(e);
+        }
+    }
+
+    [HttpGet("{goalId}/progress")]
+    [ProducesResponseType(typeof(decimal), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> GetGoalProgress(Guid userId, Guid goalId)
+    {
+        try
+        {
+            var progress = await _goalService.TrackGoalProgressAsync(userId, goalId);
+            return Ok(progress);
+        }
+        catch (Exception e)
+        {
+            return HandleException(e);
+        }
+    }
+
+    [HttpPut("{goalId}/progress")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> UpdateGoalProgress(Guid userId, Guid goalId, [FromBody] decimal amount)

```

Comment:
Modify the body of this endpoint to let know the user which values is he gonna modify, in this case increase the amount of a goal, but in the body it just ask you for a number without context